### PR TITLE
Update important-upgrade-notes.rst

### PR DESCRIPTION
**Pull request by amyblais**
- Added note for v5.35 DB migration upgrade note.
 - Added note for an upcoming deprecation in August release. **Note:** I didn't yet add the release version in case August release becomes v6.0. Hope to finalize this with PMs soon.
 - Cleaned up / re-organized some of the upgrade notes to give space for the above 2 new notes.

### CODE REVIEW

File: source/administration/important-upgrade-notes.rst [hunk 1]
```diff
@@ -5,10 +5,10 @@ Important Upgrade Notes
...
+   Due to the introduction of new features (Shared Channels and Folded Reply Threads) the performance of the migration process for the 5.35 (May 16, 2021) release has been noticeably affected. Depending on the size, type and version of the database, longer than usual upgrade times should be expected. These can vary from a couple of minutes (average case) to hours (worst case, MySQL 5.x only). A moderate to significant spike in database CPU usage should also be expected during this process. `The queries to run ahead of time to minimize the impact are provided here <https://gist.github.com/streamer45/9aee4906639a49ebde68b2f3c0f924c1>`_.
```
Comment (streamer45) [inline source/administration/important-upgrade-notes.rst:8]: Maybe I'd keep the last line more generic as the document doesn't really list queries but offers some more details on the performance impact of the migration and suggests possible mitigation strategies.

**Review by cwarnermm — COMMENTED** [review]
Proposed suggestions provided inline.

**Review by wiersgallak — APPROVED** [review]
Thanks, Amy.

**Review by streamer45 — APPROVED** [review]
Perfect :+1:

**Review by cwarnermm — APPROVED** [review]

**Pull request merged**